src/app/portlets/content-types/fields/content-type-fields-add-row/content-type-fields-add-row.component.ts

```diff
@@ -0,0 +1,188 @@
+import { Component, OnDestroy, Input, Output, EventEmitter, OnInit,
+         ViewChild, ElementRef } from '@angular/core';
+import { HotkeysService, Hotkey } from 'angular2-hotkeys';
+import { MessageService } from './../../../../api/services/messages-service';
+
+/**
+ * Display select columns row
+ *
+ * @export
+ * @class ContentTypeFieldsAddRowComponent
+ */
+
+@Component({
+    selector: 'dot-add-rows',
+    styleUrls: ['./content-type-fields-add-row.component.scss'],
+    templateUrl: './content-type-fields-add-row.component.html',
+})
+export class ContentTypeFieldsAddRowComponent implements OnDestroy, OnInit {
+    rowState = 'add';
+    selectedColumnIndex = 0;
+    i18nMessages = {};
+
+    @Input() columns: number[] = [1, 2, 3, 4];
+    @Input() disabled = false;
+    @Input() toolTips: string[] = [
+        'contenttypes.content.one_column',
+        'contenttypes.content.two_columns',
+        'contenttypes.content.three_columns',
+        'contenttypes.content.four_columns'
+    ];
+    @Output() selectColums: EventEmitter<number> = new EventEmitter<number>();
+    @ViewChild('colContainer') colContainerElem: ElementRef;
+
+    constructor(private hotkeysService: HotkeysService , public messageService: MessageService) {
+    }
+
+    ngOnInit(): void {
+        this.setKeyboardEvent('ctrl+a', this.selectColumnState.bind(this));
+        this.loadMessages();
+    }
+
+    ngOnDestroy(): void {
+        this.hotkeysService.remove(this.hotkeysService.get(['left', 'right', 'enter', 'esc']));
```

Comment:
What happen here is we try to "remove" hotkeys that doesn't exist? Because the component can be destroy and the keys are not always bind it.